src/main/java/io/lighty/yang/validator/formats/JsonTree.java

```diff
@@ -118,88 +120,115 @@ public void emitFormat() {
                 augmentationJson.put(PATH, path);
                 augmentationJson.put(NAME, path);
                 for (final DataSchemaNode child : augmentation.getChildNodes()) {
+                    schemaInferenceStack.enterSchemaTree(child.getQName());
                     if (isConfig) {
-                        augmentationJson.append(CHILDREN, resolveChildMetadata(child, Optional.empty()));
+                        augmentationJson.append(CHILDREN, resolveChildMetadata(child, Optional.empty(),
+                                schemaInferenceStack));
                     } else {
-                        augmentationJson.append(CHILDREN, resolveChildMetadata(child, Optional.of(false)));
+                        augmentationJson.append(CHILDREN, resolveChildMetadata(child, Optional.of(false),
+                                schemaInferenceStack));
                     }
+                    schemaInferenceStack.exit();
                 }
 
-                appendActionsToAugmentationJson(augmentation, augmentationJson);
-                appendNotificationsToAugmentationJson(module, augmentationJson);
+                appendActionsToAugmentationJson(augmentation, augmentationJson, schemaInferenceStack);
+                appendNotificationsToAugmentationJson(module, augmentationJson, schemaInferenceStack);
                 jsonTree.append(AUGMENTS, augmentationJson);
+                schemaInferenceStack.exit();
```

Comment:
We do not need this line - we are entering augmentation target with full path.

Reply:
Change it to `clear()`.

Reply:
Done